extensions/metrics/src/interceptors/metrics.interceptor.ts

```diff
@@ -74,22 +95,68 @@ export class MetricsInterceptor implements Provider<Interceptor> {
     next: () => ValueOrPromise<T>,
   ) {
     MetricsInterceptor.setup();
-    const endGauge = MetricsInterceptor.gauge.startTimer({
-      targetName: invocationCtx.targetName,
-    });
-    const endHistogram = MetricsInterceptor.histogram.startTimer({
+    const {request, response} = invocationCtx.parent as RequestContext;
+    const labelValues: LabelValues<LabelNames> = {
       targetName: invocationCtx.targetName,
-    });
-    const endSummary = MetricsInterceptor.summary.startTimer({
-      targetName: invocationCtx.targetName,
-    });
+      method: request.method,
+    };
+    labelValues.path = getPathFromRequest(request, invocationCtx);
+    const endGauge = MetricsInterceptor.gauge.startTimer();
+    const endHistogram = MetricsInterceptor.histogram.startTimer();
+    const endSummary = MetricsInterceptor.summary.startTimer();
     try {
-      MetricsInterceptor.counter.inc();
-      return await next();
+      const result = await next();
+      labelValues.statusCode = getStatusCodeFromResponse(response, result);
+      return result;
+    } catch (err) {
+      labelValues.statusCode = getStatusCodeFromError(err);
+      throw err;
     } finally {
-      endGauge();
-      endHistogram();
-      endSummary();
+      MetricsInterceptor.counter.inc(labelValues);
+      endGauge(labelValues);
+      endHistogram(labelValues);
+      endSummary(labelValues);
     }
   }
 }
+
+function getPathFromRequest(
+  request: Request,
+  invocationCtx: InvocationContext,
+) {
+  // replace path params with their named string placeholder
+  // this is required since every unique combination of key-value label pairs
+  // represents a new time series, which can dramatically increase the amount
+  // of data stored for values with high cardinality such as IDs
+  let {path} = request;
+  const {source} = invocationCtx;
+  if (isRouteSource(source) && isResolvedControllerRoute(source.value)) {
+    const {pathParams} = source.value;
```

Comment:
I think you should be able to get the path pattern from (source.value.path)[https://github.com/strongloop/loopback-next/blob/7f764e109a23da5e03f9a0bee93ba0c7c7d21e81/packages/rest/src/router/base-route.ts#L25], or the operationSpec.path.

Reply:
Updated it accordingly, I am using the `request.path` as fallback but I am not sure if this makes sense is it even possible right now that the invocation source can be of type `proxy`?

Reply:
The `source.type` tells if the proxy is invoked from a `route` or another class.

Reply:
but in that case which value should be used for the path?

Reply:
See https://github.com/strongloop/loopback-next/blob/7f764e109a23da5e03f9a0bee93ba0c7c7d21e81/packages/rest/src/router/base-route.ts#L47 and https://github.com/strongloop/loopback-next/blob/7f764e109a23da5e03f9a0bee93ba0c7c7d21e81/packages/context/src/interception-proxy.ts#L65

1. route - the value is RouteEntry (verb + path)
2. proxy - the value is ResolutionSession (value.toString() => binding path)

Reply:
mhm so basically if its a proxy source it would not make sense to even track the method, path, and status code